src/plugins/data/common/es_query/kuery/types.ts

```diff
@@ -17,14 +17,31 @@
  * under the License.
  */
 
-export * from './ast';
+import { NodeTypes } from './node_types';
+
+export interface KueryNode {
```

Comment:
Should we use this opportunity to rename the whole thing to KQL?
i.e. `KqlNode`, call the whole folder `kql`. etc/

@ppisljar @lukeelmers what do you think?

Reply:
Yeah, if `Kql` is the name we are sticking with (and I don't know the "official" answer to this), then it might be nice to do it all at once. But if we did that, I would recommend also updating the `esKuery` namespace to `esKql`, and changing any other references to kuery.

This could be a large-ish task, and I don't want to block progress on this PR since it is time-sensitive, so I'd defer to @Bargs or @ppisljar on whether this is worthwhile to do now, or whether we are okay possibly changing the names again in a follow-up PR.